components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfigTest.java

```diff
@@ -302,4 +303,58 @@ public void testGetExcludedParamsInActionRequestForEmptyConfigForAllAndDefinedTy
                 actionExecutorConfig.getExcludedParamsInActionRequestForActionType(ActionType.PRE_ISSUE_ACCESS_TOKEN);
         assertEquals(excludedHeaders, Collections.emptySet());
     }
+
+    @Test
+    public void testGetHttpReadTimeoutInMillis() {
+
+        Map<String, Object> configMap = new HashMap<>();
+        configMap.put("Actions.HTTPClient.HTTPReadTimeout", "5000");
+        when(mockIdentityConfigParser.getConfiguration()).thenReturn(configMap);
+        Assert.assertEquals(5000, actionExecutorConfig.getHttpReadTimeoutInMillis());
+    }
+
+    @Test
+    public void testGetHttpReadTimeoutInMillisWithoutNumberFormat() {
+
+        Map<String, Object> configMap = new HashMap<>();
+        configMap.put("Actions.HTTPClient.HTTPReadTimeout", "value");
+        when(mockIdentityConfigParser.getConfiguration()).thenReturn(configMap);
+        Assert.assertEquals(5000, actionExecutorConfig.getHttpReadTimeoutInMillis());
+    }
+
+    @Test
+    public void testGetHttpConnectionRequestTimeoutInMillis() {
+
+        Map<String, Object> configMap = new HashMap<>();
+        configMap.put("Actions.HTTPClient.HTTPConnectionRequestTimeout", "2000");
+        when(mockIdentityConfigParser.getConfiguration()).thenReturn(configMap);
+        Assert.assertEquals(2000, actionExecutorConfig.getHttpConnectionRequestTimeoutInMillis());
+    }
+
+    @Test
+    public void testGetHttpConnectionTimeoutInMillis() {
+
+        Map<String, Object> configMap = new HashMap<>();
+        configMap.put("Actions.HTTPClient.HTTPConnectionTimeout", "2000");
+        when(mockIdentityConfigParser.getConfiguration()).thenReturn(configMap);
+        Assert.assertEquals(2000, actionExecutorConfig.getHttpConnectionTimeoutInMillis());
+    }
+
+    @Test
+    public void testGetHttpConnectionPoolSize() {
+
+        Map<String, Object> configMap = new HashMap<>();
+        configMap.put("Actions.HTTPClient.HTTPConnectionPoolSize", "20");
+        when(mockIdentityConfigParser.getConfiguration()).thenReturn(configMap);
+        Assert.assertEquals(20, actionExecutorConfig.getHttpConnectionPoolSize());
+    }
+
+    @Test
+    public void testGetHttpConnectionPoolSizeWithoutNumberFormat() {
+
+        Map<String, Object> configMap = new HashMap<>();
+        configMap.put("Actions.HTTPClient.HTTPConnectionPoolSize", "value");
+        when(mockIdentityConfigParser.getConfiguration()).thenReturn(configMap);
+        Assert.assertEquals(20, actionExecutorConfig.getHttpConnectionPoolSize());
+    }
```

Comment:
Refer my comment above and fix the method name based on the test scenario you test

Reply:
Updated in https://github.com/wso2/carbon-identity-framework/pull/5935/commits/edba811642bfb3eb64866ebfef8efd9d67ae484e